src/sentry/monitors/validators.py

```diff
@@ -80,23 +87,42 @@ def validate_schedule_type(self, value):
     def validate(self, attrs):
         if "schedule_type" in attrs:
             schedule_type = attrs["schedule_type"]
+        elif self.instance:
+            schedule_type = self.instance.get("schedule_type")
         else:
-            schedule_type = self.instance["schedule_type"]
+            schedule_type = None
 
         schedule = attrs.get("schedule")
         if not schedule:
             return attrs
 
+        # Translate alternative schedule type key
+        if isinstance(schedule, dict) and schedule.get("type"):
+            schedule_type = schedule.get("type")
+            schedule_type = SCHEDULE_TYPES[schedule_type]
```

Comment:
Could you hit a KeyError here if some one put a `type` of `lol`?

Reply:
👍 I'll clean that up